k8s/upgrade/src/events/event_recorder.rs

```diff
@@ -44,14 +43,15 @@ impl EventNote {
 
 /// A builder for the Kubernetes event publisher.
 #[derive(Default)]
-pub struct EventRecorderBuilder {
+pub struct EventRecorderBuilder<'a> {
     pod_name: Option<String>,
     namespace: Option<String>,
     source_version: Option<String>,
     target_version: Option<String>,
+    product_name: &'a str,
```

Comment:
Just make it a string like everything, no point complicating the api just for this when everything else is already String

Reply:
Alright.